Build init plugin is locale sensitive

<!--- 
Please follow the instructions below. 
We receive dozens of issues every week, so to stay productive, we will close issues that don't provide enough information. 

Please open Android-related issues on the Android Issue Tracker at https://source.android.com/source/report-bugs
Please open Gradle Native-related issues at https://github.com/gradle/gradle-native/issues
-->

The Build Init plugin `--type` parameter matching is locale sensitive. When doing a `gradle init --type java-application` on a system with Turkish locale `tr-TR`, then the build fails.

### Expected Behavior

It is expected to correctly initialize a java-application.

### Current Behavior

The build fails and as it cannot match `java-application` to `java-appl?cat?on`.

```
gradle.bat init --type java-application
Picked up JAVA_TOOL_OPTIONS: -Duser.language=tr -Duser.country=TR
> Task :init FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':init'.
> The requested build type 'java-application' is not supported. Supported types:
    - 'basic'
    - 'cpp-application'
    - 'cpp-library'
    - 'groovy-appl?cat?on'
    - 'groovy-gradle-plugin'
    - 'groovy-l?brary'
    - 'java-appl?cat?on'
    - 'java-gradle-plugin'
    - 'java-l?brary'
    - 'kotlin-appl?cat?on'
    - 'kotlin-gradle-plugin'
    - 'kotlin-l?brary'
    - 'pom'
    - 'scala-appl?cat?on'
    - 'scala-l?brary'
    - 'swift-application'
    - 'swift-library'
```

### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

This is issue has been raised on Stackoverflow: [How can I initialize Gradle projects in non-English systems?](https://stackoverflow.com/q/67882533/5006866)

### Steps to Reproduce 

On Windows, 

1. Open Powershell
2. Set environment variable `JAVA_TOOL_OPTIONS` and configure locale `tr-TR`
    ```
    $env:JAVA_TOOL_OPTIONS="-Duser.language=tr -Duser.country=TR"
    ```
3. Execute `gradle.bat init --type java-application`

<!--- Provide a self-contained example project (as an attached archive or a Github project). -->
<!--- In the rare cases where this is infeasible, we will also accept a detailed set of instructions. -->

### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- A build scan https://scans.gradle.com/get-started is ideal -->
* Windows 10
* Turkish locale
* Powershell

--- a/subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/JavaApplicationInitIntegrationTest.groovy
+++ b/subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/JavaApplicationInitIntegrationTest.groovy
@@ -19,6 +19,9 @@ package org.gradle.buildinit.plugins
 import org.gradle.buildinit.plugins.fixtures.ScriptDslFixture
 import org.gradle.buildinit.plugins.internal.modifiers.BuildInitTestFramework
 import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
+import org.gradle.integtests.fixtures.executer.GradleContextualExecuter
+import spock.lang.IgnoreIf
+import spock.lang.Issue
 import spock.lang.Unroll
 
 import static org.gradle.buildinit.plugins.internal.modifiers.BuildInitDsl.GROOVY
@@ -224,5 +227,16 @@ class JavaApplicationInitIntegrationTest extends AbstractInitIntegrationSpec {
 
         where:
         scriptDsl << ScriptDslFixture.SCRIPT_DSLS
+    }
+
+
+    @Issue("https://github.com/gradle/gradle/issues/17383")
+    @IgnoreIf({ GradleContextualExecuter.embedded })
+    def "command line works with different locale"() {
+        setup:
+        executer.withCommandLineGradleOpts('-Duser.language=tr', '-Duser.country=TR')
+
+        expect:
+        succeeds('init', '--type', 'java-application', '--dsl', 'groovy')
     }
 }

--- a/subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/modifiers/Names.java
+++ b/subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/modifiers/Names.java
@@ -18,14 +18,16 @@ package org.gradle.buildinit.plugins.internal.modifiers;
 
 import com.google.common.base.Joiner;
 
+import java.util.Locale;
+
 class Names {
     static String displayNameFor(Enum<?> value) {
-        String[] parts = value.name().toLowerCase().split("_");
+        String[] parts = value.name().toLowerCase(Locale.US).split("_");
         return Joiner.on(' ').join(parts);
     }
 
     static String idFor(Enum<?> value) {
-        String[] parts = value.name().toLowerCase().split("_");
+        String[] parts = value.name().toLowerCase(Locale.US).split("_");
         return Joiner.on('-').join(parts);
     }
 }

